pkg/scheduler/scheduler.go

```diff
@@ -0,0 +1,89 @@
+package scheduler
+
+import (
+	"context"
+	"github.com/kyma-incubator/octopus/pkg/apis/testing/v1alpha1"
+	"github.com/kyma-incubator/octopus/pkg/consts"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate mockery -name=StatusProvider -output=automock -outpkg=automock -case=underscore
```

Comment:
for someone who will execute `make generate` and do not have mockery, he will receive such err:
```
pkg/scheduler/scheduler.go:13: running "mockery": exec: "mockery": executable file not found in $PATH
```

`generate` target is used also in other targets like `test` etc.

in prow-addons-ctr this issue is resolved, u can check this makefile: https://github.com/kyma-project/test-infra/blob/master/development/prow-addons-ctrl-manager/Makefile

or u can come up sth different, important is that it will work for od CI and for other developers :)

Reply:
I modified comment and use mockery from vendor